### PR TITLE
Changes to pattern folder name to remove construct from it and referencing correct path in .ts files under bin and respective changes in README.md file

### PR DESCRIPTION
**Pull request by naveen-bathula**
Changes to pattern folder name to remove construct from it and referencing correct path in .ts files under bin and respective changes in README.md file

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: bin/existing-eks-awsnative-observability.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import ExistingEksAwsNativeObservabilityConstruct from '../lib/existing-eks-awsnative-observability-construct';
+import ExistingEksAwsNativeObservabilityConstruct from '../lib/existing-eks-awsnative-observability';
```
Comment (elamaran11) [inline bin/existing-eks-awsnative-observability.ts:1]: Why is `ExistingEksAwsNativeObservabilityConstruct` still having `Construct` ? We want to totally get rid of word Construct with `Pattern`. Also we dont want `-construct` to be removed. We would rather want it to be renamed to `-pattern`.

File: README.md [hunk 2]
```diff
@@ -207,13 +207,13 @@ The `compile` command is optimized to build only modified files and is fast.
...
+1. Under lib create a folder for your pattern, such as `<pattern-name>`. If you plan to create a set of patterns that represent a particular subdomain, e.g. `security` or `hardening`, please create an issue to discuss it first. If approved, you will be able to create a folder with your subdomain name and group your pattern constructs under it. 
 2. Blueprints generally don't require a specific class, however we use a convention of wrapping each pattern in a plain class like `<Pattern-Name>Construct`. This class is generally placed in `index.ts` under your pattern folder. 
 3. Once the pattern implementation is ready, you need to include it in the list of the patterns by creating a file `bin/<pattern-name>.ts`. The implementation of this file is very light, and it is done to allow patterns to run independently.
 
 Example simple synchronous pattern:
 ```typescript
-import SingleNewEksOpenSourceobservabilityConstruct from '../lib/single-new-eks-opensource-observability-construct';
```
Comment (elamaran11) [inline README.md:216]: Why is `SingleNewEksOpenSourceobservabilityConstruct` still having `Construct` ? We want to totally get rid of word Construct with `Pattern`. Also we dont want `-construct` to be removed. We would rather want it to be renamed to `-pattern`.

File: bin/existing-eks-mixed-observability.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import ExistingEksMixedConstruct from '../lib/existing-eks-mixed-observability-construct';
+import ExistingEksMixedConstruct from '../lib/existing-eks-mixed-observability';
```
Comment (elamaran11) [inline bin/existing-eks-mixed-observability.ts:1]: Why is class name `ExistingEksMixedConstruct ` still having `Construct` ? We want to totally get rid of word Construct with `Pattern`. Also we dont want `-construct` to be removed. We would rather want it to be renamed to `-pattern`.

File: bin/existing-eks-opensource-observability.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import ExistingEksOpenSourceobservabilityConstruct from '../lib/existing-eks-opensource-observability-construct';
+import ExistingEksOpenSourceobservabilityConstruct from '../lib/existing-eks-opensource-observability';
```
Comment (elamaran11) [inline bin/existing-eks-opensource-observability.ts:1]: Why is class name `ExistingEksOpenSourceobservabilityConstruct`  still having Construct ? We want to totally get rid of word Construct with Pattern. Also we dont want -construct to be removed. We would rather want it to be renamed to -pattern.

File: bin/single-new-eks-awsnative-observability.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import SingleNewEksAWSNativeobservabilityConstruct from '../lib/single-new-eks-awsnative-observability-construct';
+import SingleNewEksAWSNativeobservabilityConstruct from '../lib/single-new-eks-awsnative-observability';
```
Comment (elamaran11) [inline bin/single-new-eks-awsnative-observability.ts:1]: Why is class name `SingleNewEksAWSNativeobservabilityConstruct` still having Construct ? We want to totally get rid of word Construct with Pattern. Also we dont want -construct to be removed. We would rather want it to be renamed to -pattern.

File: bin/single-new-eks-graviton-opensource-observability.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import { configureApp } from '../lib/common/construct-utils';
-import SingleNewEksGravitonOpenSourceObservabilityConstruct from '../lib/single-new-eks-opensource-observability-construct/graviton-index';
+import SingleNewEksGravitonOpenSourceObservabilityConstruct from '../lib/single-new-eks-opensource-observability/graviton-index';
```
Comment (elamaran11) [inline bin/single-new-eks-graviton-opensource-observability.ts:2]: Why is class name `SingleNewEksGravitonOpenSourceObservabilityConstruct ` still having Construct ? We want to totally get rid of word Construct with Pattern. Also we dont want -construct to be removed. We would rather want it to be renamed to -pattern.

File: bin/single-new-eks-mixed-observability.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import SingleNewEksMixedobservabilityConstruct from '../lib/single-new-eks-mixed-observability-construct';
+import SingleNewEksMixedobservabilityConstruct from '../lib/single-new-eks-mixed-observability';
```
Comment (elamaran11) [inline bin/single-new-eks-mixed-observability.ts:1]: Why is class name `SingleNewEksMixedobservabilityConstruct ` still having Construct ? We want to totally get rid of word Construct with Pattern. Also we dont want -construct to be removed. We would rather want it to be renamed to -pattern.

File: bin/single-new-eks-opensource-observability.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import SingleNewEksOpenSourceobservabilityConstruct from '../lib/single-new-eks-opensource-observability-construct';
+import SingleNewEksOpenSourceobservabilityConstruct from '../lib/single-new-eks-opensource-observability';
```
Comment (elamaran11) [inline bin/single-new-eks-opensource-observability.ts:1]: Why is class name `SingleNewEksOpenSourceobservabilityConstruct ` still having Construct ? We want to totally get rid of word Construct with Pattern. Also we dont want -construct to be removed. We would rather want it to be renamed to -pattern.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@naveen-bathula Thankyou for the PR. Could you please rename construct in all lib ts files with pattern than dropping it. Also all class names should have construct replaced with pattern. Also please review two patterns and provide testing screenshots. I prefer AWS Native new cluster and Open source existing cluster make sure to update docs if they are impacted for any names from make command outputs.

**Comment by naveen-bathula** [comment]
> @naveen-bathula Thankyou for the PR. Could you please rename construct in all lib ts files with pattern than dropping it. Also all class names should have construct replaced with pattern. Also please review two patterns and provide testing screenshots. I prefer AWS Native new cluster and Open source existing cluster make sure to update docs if they are impacted for any names from make command outputs.

@elamaran11 All the requested changes have been made and two suggested patterns were tested and screenshots attached.

CloudWatch1
CloudWatch2
CloudWatch3
![Grafana2

Grafana1
](https://github.com/aws-observability/cdk-aws-observability-accelerator/assets/32726946/6012ff6f-a2c0-451f-916b-77864626c94c)

**Comment by elamaran11** [comment]
@shapirov103 Can you quickly check if this is looking good for issue we had.

**Review by elamaran11 — APPROVED** [review]
LGTM.

**Review by shapirov103 — APPROVED** [review]
LGTM

@naveen-bathula I suppose you captured all the doc changes as well. It sounds a bit odd that there ~10 patterns and only two changes in the docs.

**Comment by shapirov103** [comment]
@naveen-bathula please address the github action error 
https://github.com/aws-observability/cdk-aws-observability-accelerator/actions/runs/5907300979/job/16085531340?pr=91

```
rm -rf dist && node node_modules/.bin/tsc --skipLibCheck
Error: lib/existing-eks-opensource-observability-pattern/index.ts(6[5](https://github.com/aws-observability/cdk-aws-observability-accelerator/actions/runs/5907300979/job/16085531340?pr=91#step:7:6),17): error TS2345: Argument of type '{ bootstrapRepo: { repoUrl: string; name: string; targetRevision: string; path: string; }; fluxTargetNamespace: string; bootstrapValues: { AMG_AWS_REGION: string; AMP_ENDPOINT_URL: string; AMG_ENDPOINT_URL: string | undefined; ... 5 more ...; GRAFANA_WORKLOADS_DASH_URL: string; }; }' is not assignable to parameter of type 'FluxCDAddOnProps'.
  Object literal may only specify known properties, and 'bootstrapRepo' does not exist in type 'FluxCDAddOnProps'.
Error: lib/single-new-eks-opensource-observability-pattern/graviton-index.ts(49,17): error TS2345: Argument of type '{ bootstrapRepo: { repoUrl: string; name: string; targetRevision: string; path: string; }; fluxTargetNamespace: string; bootstrapValues: { AMG_AWS_REGION: string; AMP_ENDPOINT_URL: string; AMG_ENDPOINT_URL: string | undefined; ... 5 more ...; GRAFANA_WORKLOADS_DASH_URL: string; }; }' is not assignable to parameter of type 'FluxCDAddOnProps'.
  Object literal may only specify known properties, and 'bootstrapRepo' does not exist in type 'FluxCDAddOnProps'.
Error: lib/single-new-eks-opensource-observability-pattern/index.ts(51,1[7](https://github.com/aws-observability/cdk-aws-observability-accelerator/actions/runs/5907300979/job/16085531340?pr=91#step:7:8)): error TS2345: Argument of type '{ bootstrapRepo: { repoUrl: string; name: string; targetRevision: string; path: string; }; fluxTargetNamespace: string; bootstrapValues: { AMG_AWS_REGION: string; AMP_ENDPOINT_URL: string; AMG_ENDPOINT_URL: string | undefined; ... 5 more ...; GRAFANA_WORKLOADS_DASH_URL: string; }; }' is not assignable to parameter of type 'FluxCDAddOnProps'.
  Object literal may only specify known properties, and 'bootstrapRepo' does not exist in type 'FluxCDAddOnProps'.
make: *** [build] Error 2
Error: Process completed with exit code 2.
```

**Comment by elamaran11** [comment]
> LGTM
> 
> @naveen-bathula I suppose you captured all the doc changes as well. It sounds a bit odd that there ~10 patterns and only two changes in the docs.

This is because we are not changing the name of the pattern only the file names and class names. @naveen-bathula Please double confirm on Mikhail's feedback.

**Comment by elamaran11** [comment]
> @naveen-bathula please address the github action error https://github.com/aws-observability/cdk-aws-observability-accelerator/actions/runs/5907300979/job/16085531340?pr=91
> 
> ```
> rm -rf dist && node node_modules/.bin/tsc --skipLibCheck
> Error: lib/existing-eks-opensource-observability-pattern/index.ts(6[5](https://github.com/aws-observability/cdk-aws-observability-accelerator/actions/runs/5907300979/job/16085531340?pr=91#step:7:6),17): error TS2345: Argument of type '{ bootstrapRepo: { repoUrl: string; name: string; targetRevision: string; path: string; }; fluxTargetNamespace: string; bootstrapValues: { AMG_AWS_REGION: string; AMP_ENDPOINT_URL: string; AMG_ENDPOINT_URL: string | undefined; ... 5 more ...; GRAFANA_WORKLOADS_DASH_URL: string; }; }' is not assignable to parameter of type 'FluxCDAddOnProps'.
>   Object literal may only specify known properties, and 'bootstrapRepo' does not exist in type 'FluxCDAddOnProps'.
> Error: lib/single-new-eks-opensource-observability-pattern/graviton-index.ts(49,17): error TS2345: Argument of type '{ bootstrapRepo: { repoUrl: string; name: string; targetRevision: string; path: string; }; fluxTargetNamespace: string; bootstrapValues: { AMG_AWS_REGION: string; AMP_ENDPOINT_URL: string; AMG_ENDPOINT_URL: string | undefined; ... 5 more ...; GRAFANA_WORKLOADS_DASH_URL: string; }; }' is not assignable to parameter of type 'FluxCDAddOnProps'.
>   Object literal may only specify known properties, and 'bootstrapRepo' does not exist in type 'FluxCDAddOnProps'.
> Error: lib/single-new-eks-opensource-observability-pattern/index.ts(51,1[7](https://github.com/aws-observability/cdk-aws-observability-accelerator/actions/runs/5907300979/job/16085531340?pr=91#step:7:8)): error TS2345: Argument of type '{ bootstrapRepo: { repoUrl: string; name: string; targetRevision: string; path: string; }; fluxTargetNamespace: string; bootstrapValues: { AMG_AWS_REGION: string; AMP_ENDPOINT_URL: string; AMG_ENDPOINT_URL: string | undefined; ... 5 more ...; GRAFANA_WORKLOADS_DASH_URL: string; }; }' is not assignable to parameter of type 'FluxCDAddOnProps'.
>   Object literal may only specify known properties, and 'bootstrapRepo' does not exist in type 'FluxCDAddOnProps'.
> make: *** [build] Error 2
> Error: Process completed with exit code 2.
> ```

@naveen-bathula Lets this fix since you are already having an active PR. Check [this](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/docs/addons/fluxcd.md#single-bootstrap-repo-path) doc for an example to fix it. This is just parameter fix and you can turn it on quickly and just do a quick test. This is because of blueprints flux change breaking the accelerator. Please fix this asap.

File: lib/existing-eks-opensource-observability-pattern/index.ts [hunk 9]
```diff
@@ -62,31 +62,36 @@ export default class ExistingEksOpenSourceobservabilityConstruct {
...
+                                "AMP_ENDPOINT_URL": ampPrometheusEndpoint,
+                                "AMG_ENDPOINT_URL": amgEndpointUrl,
+                                "GRAFANA_CLUSTER_DASH_URL" : clusterDashUrl,
+                                "GRAFANA_KUBELET_DASH_URL" : kubeletDashUrl,
+                                "GRAFANA_NSWRKLDS_DASH_URL" : namespaceWorkloadsDashUrl,
+                                "GRAFANA_NODEEXP_DASH_URL" : nodeExporterDashUrl,
+                                "GRAFANA_NODES_DASH_URL" : nodesDashUrl,
+                                "GRAFANA_WORKLOADS_DASH_URL" : workloadsDashUrl
+                        },
+                        kustomizations: [{kustomizationPath:"./artifacts/grafana-operator-manifests/eks/infrastructure"}]
```
Comment (elamaran11) [inline lib/existing-eks-opensource-observability-pattern/index.ts:84]: You are missing `kustomizationTargetNamespace` you should have `            kustomizations: [{kustomizationPath:"./artifacts/grafana-operator-manifests/eks/infrastructure",kustomizationTargetNamespace: "grafana-operator"}]`

File: lib/single-new-eks-opensource-observability-pattern/graviton-index.ts [hunk 10]
```diff
@@ -46,24 +46,28 @@ export default class SingleNewEksGravitonOpenSourceObservabilityConstruct {
...
+                                "AMP_ENDPOINT_URL": ampPrometheusEndpoint,
+                                "AMG_ENDPOINT_URL": amgEndpointUrl,
+                                "GRAFANA_CLUSTER_DASH_URL" : clusterDashUrl,
+                                "GRAFANA_KUBELET_DASH_URL" : kubeletDashUrl,
+                                "GRAFANA_NSWRKLDS_DASH_URL" : namespaceWorkloadsDashUrl,
+                                "GRAFANA_NODEEXP_DASH_URL" : nodeExporterDashUrl,
+                                "GRAFANA_NODES_DASH_URL" : nodesDashUrl,
+                                "GRAFANA_WORKLOADS_DASH_URL" : workloadsDashUrl
+                        },
+                        kustomizations: [{kustomizationPath:"./artifacts/grafana-operator-manifests/eks/infrastructure"}]
```
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-pattern/graviton-index.ts:68]: You are missing kustomizationTargetNamespace you should have             kustomizations: [{kustomizationPath:"./artifacts/grafana-operator-manifests/eks/infrastructure",kustomizationTargetNamespace: "grafana-operator"}]

File: lib/single-new-eks-opensource-observability-pattern/index.ts [hunk 11]
```diff
@@ -48,31 +48,36 @@ export default class SingleNewEksOpenSourceobservabilityConstruct {
...
+                                "AMP_ENDPOINT_URL": ampPrometheusEndpoint,
+                                "AMG_ENDPOINT_URL": amgEndpointUrl,
+                                "GRAFANA_CLUSTER_DASH_URL" : clusterDashUrl,
+                                "GRAFANA_KUBELET_DASH_URL" : kubeletDashUrl,
+                                "GRAFANA_NSWRKLDS_DASH_URL" : namespaceWorkloadsDashUrl,
+                                "GRAFANA_NODEEXP_DASH_URL" : nodeExporterDashUrl,
+                                "GRAFANA_NODES_DASH_URL" : nodesDashUrl,
+                                "GRAFANA_WORKLOADS_DASH_URL" : workloadsDashUrl
+                        },
+                        kustomizations: [{kustomizationPath:"./artifacts/grafana-operator-manifests/eks/infrastructure"}]
```
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-pattern/index.ts:70]: You are missing kustomizationTargetNamespace you should have             kustomizations: [{kustomizationPath:"./artifacts/grafana-operator-manifests/eks/infrastructure",kustomizationTargetNamespace: "grafana-operator"}]

File: package.json [hunk 12]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cdk-aws-observability-accelerator",
-  "version": "0.1.0",
+  "version": "1.11.0",
```
Comment (elamaran11) [inline package.json:3]: this should not change

File: package.json [hunk 13]
```diff
@@ -27,13 +27,13 @@
   "dependencies": {
     "@aws-quickstart/eks-blueprints": "^1.10.0",
     "constructs": "^10.0.0",
-    "aws-cdk": "2.86.0",
+    "aws-cdk": "2.91.0",
     "eks-blueprints-cdk-kubeflow-ext": "0.1.9",
     "source-map-support": "^0.5.21"
   },
   "overrides": {
     "@aws-quickstart/eks-blueprints": "^1.10.0",
```
Comment (elamaran11) [inline package.json:35]: this should be changed to `"^1.11.0`

File: package.json [hunk 14]
```diff
@@ -27,13 +27,13 @@
   "dependencies": {
     "@aws-quickstart/eks-blueprints": "^1.10.0",
```
Comment (elamaran11) [inline package.json:28]: this should be changed to `"^1.11.0`

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@naveen-bathula Thankyou for doing this. Some comments. One important change is missing

**Review by elamaran11 — DISMISSED** [review]
LGTM. Lets show evidence of working grafana boards, so we can merge.

**Comment by naveen-bathula** [comment]
> LGTM. Please let me know once your e2e is working @naveen-bathula We can merge it.

@elamaran11 

I have tested single-new-eks-opensource-observability and existing-eks-opensource-observability patterns, attached screenshots

Grafana3

Grafana4

**Comment by elamaran11** [comment]
Nice work @naveen-bathula, Merged.

**Pull request merged**